kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/config/BigQuerySinkConfig.java

```diff
@@ -512,133 +536,145 @@ public static ConfigDef getConfig() {
+          "The connector config is invalid and contains the following errors:\n"
+              + String.join("\n", errors)
+      );
     }
+  }
 
-  @SuppressWarnings("unchecked")
-  public static class Validator implements ConfigDef.Validator {
+  public static class MapValidator implements ConfigDef.Validator {
     @Override
     public void ensureValid(String name, Object value) {
-      switch (name) {
-        case DATASETS_CONFIG:
-          ensureValidMap(name, (List<String>) value);
-          break;
-        case TOPICS_TO_TABLES_CONFIG:
-          ensureValidMap(name, (List<String>) value);
-          break;
-        default:
-          break;
-      }
-    }
-
-    protected static void ensureValidMap(String name, List<String> values) {
-      if (values == null) {
+      if (value == null) {
         return;
       }
-      values.forEach((entry) -> parseMapping(entry, name));
+
+      @SuppressWarnings("unchecked")
+      List<String> parsedValue = (List<String>) value;
+
+      parsedValue.forEach(BigQuerySinkConfig::parseMapping);
     }
 
-    /**
-    * Ensures the mapping given is valid, then returns an entry containing its key and value.
-    * Checks to make sure that the given String adheres to the specified format, and throws
-    * an exception if it does not. Trims leading and trailing whitespace, and then checks to make
-    * sure that both Strings are still non-empty.
-    *
-    * @param mapping The mapping to parse (should be of the form &lt;key&gt;=&lt;value&gt;)
-    * @param name The name of the field. Used in error messages.
-    * @return A Map.Entry containing the parsed key/value pair.
-    */
-    protected static Map.Entry<String, String> parseMapping(String mapping, String name) {
-      String[] keyValue = mapping.split("=");
-      if (keyValue.length != 2) {
-        throw new ConfigException(
-            "Invalid mapping for " + name
-            + " property: '" + mapping
-            + "' (must follow format '<key>=<value>')"
-        );
-      }
+    @Override
+    public String toString() {
+      return "A list of key-value pairs in the format <key1>=<value1>, <key2>=<value2>, ...";
+    }
+  }
 
-      String key = keyValue[0].trim();
-      if (key.isEmpty()) {
-        throw new ConfigException(
-            "Empty key found in mapping '" + mapping
-            + "' for " + name + " property"
-        );
-      }
+  /**
+   * Ensures the mapping given is valid, then returns an entry containing its key and value.
+   * Checks to make sure that the given String adheres to the specified format, and throws
+   * an exception if it does not. Trims leading and trailing whitespace, and then checks to make
+   * sure that both Strings are still non-empty.
+   *
+   * @param mapping The mapping to parse (should be of the form &lt;key&gt;=&lt;value&gt;)
+   * @return A Map.Entry containing the parsed key/value pair.
+   */
+  static Map.Entry<String, String> parseMapping(String mapping) {
```

Comment:
Is there a reason `name` is removed from the definition?

Reply:
It makes the method cleaner (easier to use with method reference syntax, for example, like is done in the `MapValidator` class), and it's unnecessary since when the method is invoked in the `MapValidator`, any errors that come up from it will automatically be associated with the property that's being validated and displayed to the user in the REST API response for the request to create/reconfigure the connector or validate a config for it.